### DISCUSSION TITLE
How to make changes in the Circle application on the fly?

### DISCUSSION THREAD

**Discussion by yashtandon113** [ant-media/Ant-Media-Server#6335] [FAQs]
I have already installed the .war file for the Circle app on my AMS.

Now, I would like to update the interface of the application. The method you suggested earlier seems to apply to a new installation by cloning the repo and building a new war file, but I need to modify an existing installation. Could you provide the steps for this, please?

When I have new changes every time, do I need to generate war files every time and always create new applications?

**Comment by zerostack0** [comment]
You have 2 options. If you want to use the Circle on the Ant Media Server, here are the steps you need to follow every time you need to update the application:
- Generate a new war file.
- Remove the old application on the management console.
- Upload the new war file.

The other option is after you install the Circle once, you can follow the steps below:
- Set the WebSocket URL in environment variables with your instance. Checkout here: https://github.com/ant-media/conference-call-application/blob/main/react/.env.production#L13
- Run  `npm install` inside the react folder.
- Run `npm run build` command
- You can just serve the build folder in any HTTP server. It's much more easier to update it on the fly.
